oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/authentication/OidcIdTokenValidator.java

```diff
@@ -143,6 +146,13 @@ public void setClock(Clock clock) {
 		this.clock = clock;
 	}
 
+	private boolean issuerMatchesMetadata(Jwt idToken) {
+		String metadataIssuer = (String) clientRegistration.getProviderDetails().getConfigurationMetadata()
+				.get("issuer");
```

Comment:
The issuer claim is `iss`. You can use `JwtClaimNames.ISS`.

Reply:
Are you sure? The claim is called `issuer` in the provider metadata and `iss` in the id token (https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata).

Reply:
Of course, my mistake. Thanks for correcting me @mengelbrecht !